go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -3970,6 +3973,61 @@ func (s *VtctldServer) SetShardTabletControl(ctx context.Context, req *vtctldata
 	}, nil
 }
 
+// SetVtorcEmergencyReparent enable/disables the use of EmergencyReparentShard in VTOrc recoveries for a given keyspace or keyspace/shard.
+func (s *VtctldServer) SetVtorcEmergencyReparent(ctx context.Context, req *vtctldatapb.SetVtorcEmergencyReparentRequest) (resp *vtctldatapb.SetVtorcEmergencyReparentResponse, err error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.SetVtorcEmergencyReparent")
+	defer span.Finish()
+
+	defer panicHandler(&err)
+
+	span.Annotate("keyspace", req.Keyspace)
+	span.Annotate("shard", req.Shard)
+	span.Annotate("disable", req.Disable)
+
+	ctx, unlock, lockErr := s.ts.LockKeyspace(ctx, req.Keyspace, "SetVtorcEmergencyReparent")
+	if lockErr != nil {
+		err = lockErr
+		return nil, err
+	}
+
+	defer unlock(&err)
+
+	if req.Shard != "" && req.Shard != "-" {
```

Comment:
@GuptaManan100 I may have got `req.Shard != "-"` backwards, what I intend here is if a keyspace is unsharded we just disable ERS for the whole keyspace, because there will never be more shards

I think I probably meant to use `req.Shard != "0"` here, Can you confirm?

Reply:
We allow both `-` and `0` for unsharded keyspaces, so you would have to check both.

Reply:
@deepthi / @GuptaManan100 updated the logic to my thinking above

If a `-` shard cannot be split then I believe both should be checked, but if it CAN be split, I'd prefer the state to not be inherited by the destination shards, which I believe my recent commit achieves